config/main.go

```diff
@@ -37,3 +44,93 @@ var (
 	// if RequirePass is set to an empty string, no authentication is required
 	RequirePass string = constants.EmptyStr
 )
+
+func init() {
+	args := os.Args[1:]
+
+	if len(args) == 1 && args[0] == constants.InitConfig {
+
+		err := os.MkdirAll(filepath.Dir(ConFilePath), 0755)
+		if err != nil {
+			if os.IsPermission(err) {
+				log.Error("Permission denied to create directory")
+			}
+			log.Errorf("Failed to create directory: %v", err)
+		}
+
+		if _, err := os.Stat(ConFilePath); err == nil {
+			log.Infof("Config file already exists at %s", ConFilePath)
+			ParseConfig(ConFilePath)
+			return
+		}
+
+		configFile, err := os.OpenFile(ConFilePath, os.O_CREATE|os.O_RDWR, 0666)
+		if err != nil {
+			log.Fatalf("Failed to create config file: %v", err.Error())
+		}
+
+		log.Infof("Creating default config file at %s", ConFilePath)
+
+		// copy the default config file to the path default file is ./config/dice.conf
+		data, err := os.ReadFile("./dice.conf")
+		if err != nil {
+			log.Fatalf("Failed to read default config file: %v", err.Error())
+		}
+		_, err = configFile.Write(data)
+		if err != nil {
+			log.Fatalf("Failed to write to config file: %v", err.Error())
```

Comment:
Not sure if these errors should be fatal? Unless we're able to run on defaults we should warn the user and continue I believe if any explicit config file path hasn't been passed by user to be used.